src/controlledorbitalcamera.h

```diff
@@ -0,0 +1,56 @@
+/* OpenAWE - A reimplementation of Remedys Alan Wake Engine
+ *
+ * OpenAWE is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * OpenAWE is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * OpenAWE is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with OpenAWE. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef OPENAWE_CONTROLLEDORBITALCAMERA_H
+#define OPENAWE_CONTROLLEDORBITALCAMERA_H
+
+#include <memory>
+
+#include <bullet/btBulletDynamicsCommon.h>
+
+#include "src/events/event.h"
+
+#include "src/graphics/orbitalcamera.h"
+
+#include "src/physics/charactercontroller.h"
+
+class ControlledOrbitalCamera : public Graphics::OrbitalCamera {
+public:
+	ControlledOrbitalCamera();
+
+	void attachTo(Physics::CharacterControllerPtr object);
+
+	void update(float delta) override;
+
+private:
+	void handleRotation(const Events::Event &event);
+	void handleStates(const Events::Event &event);
+
+	float _mouseSensitivity = 1.0f, _gamepadSensitivity = 10.0f;
+
+	Physics::CharacterControllerPtr _followedObject;
+
+	btSphereShape _castSphere;
```

Comment:
The idea of this was to avoid frequent creation-destruction of bullet shapes for handling collisions (especially because it happens every frame for the orbital camera) by storing an object used for collision inside the camera and passing it as a parameter. This could be instead mitigated by storing a sphere object in `PhysicsManager` (which seemed inappropriate at the time as you would have to keep changing sphere's radius for each cast) or by creating an object pool for the shapes.